python_files/ml_on_iris_dataset.py

```diff
@@ -0,0 +1,192 @@
+    fig.show()
+
+    # strip plot
+    fig = px.strip(data_df, x="petal_len_cm", y="class", orientation="h", color="class")
+    fig.show()
+
+    # parallel coordinates plot
+    # note: ['Iris-setosa': 0, 'Iris-versicolor':1, 'Iris-virginica':2]
+    plant_names, plant_indx = np.unique(data_df["class"], return_inverse=True)
+    fig = px.parallel_coordinates(
+        data_df,
+        color=plant_indx,
+        labels={
+            "color": "Plant",
+            "sepal_wid_cm": "Sepal Width",
+            "sepal_leng_cm": "Sepal Length",
+            "petal_wid_cm": "Petal Width",
+            "petal_len_cm": "Petal Length",
+        },
+        color_continuous_scale=px.colors.diverging.Tealrose,
+        color_continuous_midpoint=1,
+    )
+
+    fig.update_layout(
+        coloraxis_colorbar=dict(
+            title="Plant", tickvals=list(set(plant_indx)), ticktext=plant_names
+        )
+    )
+    fig.show()
+
+    # adds pipeline
+    print_heading("Creating Pipelines")
+    rf_pipeline = Pipeline(
+        steps=[
+            ("Normalizer", Normalizer()),
+            ("RandomForest", RandomForestClassifier(random_state=0)),
+        ]
+    )
+
+    svc_pipeline = Pipeline(
+        steps=[
+            ("Normalizer", Normalizer()),
+            ("OneHotEncoder", OneHotEncoder()),
+            ("OneVsRestClassifer", LinearSVC(random_state=0)),
+        ]
+    )
+    print(rf_pipeline)
+    print(svc_pipeline)
+
+    print_heading("Pipeline Peformance")
+    rf_pipeline.fit(X_orig, y)
+    svc_pipeline.fit(X_orig, y)
+    rf_probs = rf_pipeline.predict_proba(X_orig)
+    rf_preds = rf_pipeline.predict(X_orig)
+    svc_preds = svc_pipeline.predict(X_orig)
+
+    print("Overall Performance")
+    print("RandomForest Log-Loss:  %.3f" % log_loss(y, rf_probs))
+    print("RandomForest Score: %.3f" % rf_pipeline.score(X_orig, y))
+    print("LinearSVC Score: %.3f" % svc_pipeline.score(X_orig, y))
+    print("", end="\n")
+
+    # encoded target col and preds to get area under ROC Curve
+    y_trans_encoded = OneHotEncoder(sparse=False).fit_transform(y.reshape(-1, 1))
+    svc_preds_trans = OneHotEncoder(sparse=False).fit_transform(
+        svc_preds.reshape(-1, 1)
+    )
+    rf_preds_trans = OneHotEncoder(sparse=False).fit_transform(rf_preds.reshape(-1, 1))
+
+    # calculates area under ROC curve for each target (3) and prints their performance
+    print("Each Target's Performance (Area Under ROC Curve)")
+    for i in range(0, len(iris_plants)):
+        roc = roc_auc_score(
+            y_trans_encoded[:, i], rf_preds_trans[:, i], multi_class="ovr"
+        )
+        print(
+            "Area under ROC Curve for %s Using RandomForest: %.3f"
+            % (iris_plants[i], roc)
+        )
+
+        roc = roc_auc_score(
+            y_trans_encoded[:, i], svc_preds_trans[:, i], multi_class="ovr"
+        )
+        print(
+            "Area under ROC Curve for %s Using LinearSVC: %.3f" % (iris_plants[i], roc)
```

Comment:
Felt like you should've used one of your 5 plots for plotting the ROC curve. That would have been neat to see this alongside your table results.

Reply:
I'll add that :)